tests/integration/test_api_endpoints.py

```
@@ -0,0 +1,488 @@
+class TestAPIEndpoints:
+    """Test suite for TransDock API endpoints."""
+
+    @pytest.fixture
+    def client(self):
+        """Create test client."""
+        return TestClient(app)
+
+    @pytest.fixture
+    def mock_migration_service(self):
+        """Mock migration service."""
+        from unittest.mock import AsyncMock
```

Ruff (0.11.9)

Redefinition of unused `AsyncMock` from line 10

Remove definition: `AsyncMock`

(F811)